File: src/main/java/io/pivotal/example/restcallscheduled/ScheduledMethods.java
package io.pivotal.example.restcallscheduled;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;
import org.springframework.web.client.RestTemplate;

@Component
public class ScheduledMethods {

    private final Logger LOG = LoggerFactory.getLogger(this.getClass());

    private final RestTemplate restTemplate;
    @Value("${rest.url}")
    private String restUrl;


    public ScheduledMethods(RestTemplate restTemplate) {
        this.restTemplate = restTemplate;
    }

    @Scheduled(cron = "${cron.job.schedule}")
    public void makeRestCall() {
        String response = restTemplate.getForObject(restUrl, String.class);
        LOG.debug(response);
    }
}

File: src/main/kotlin/io/pivotal/example/restcallscheduled/ScheduledMethods.kt
package io.pivotal.example.restcallscheduled

import org.springframework.beans.factory.annotation.Value
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component
import org.springframework.web.client.RestTemplate

@Component
class ScheduledMethods(private val restTemplate: RestTemplate) {

    @Value("\${rest.url}")
    lateinit var restUrl: String

    @Scheduled(cron = "\${cron.job.schedule}")
    fun makeRestCall() {
        val response = restTemplate.getForObject(restUrl, String::class.java)
        println(response)
    }
}
